DNN Platform/Dnn.AuthServices.Jwt/Components/Common/Controllers/JwtController.cs

```diff
@@ -224,17 +223,29 @@ public LoginResultData RenewToken(HttpRequestMessage request, string renewalToke
                 return EmptyWithError("not-more-renewal");
             }
 
+            if (ptoken.RenewalHash != GetHashedStr(renewalToken))
+            {
+                if (Logger.IsTraceEnabled) Logger.Trace("Invalid renewal token");
+                return EmptyWithError("bad-renewal-token");
+            }
+
+            if (ptoken.TokenHash != GetHashedStr(rawToken))
+            {
+                if (Logger.IsTraceEnabled) Logger.Trace("Invalid access token");
+                return EmptyWithError("bad-access-token");
+            }
+
             var userInfo = TryGetUser(jwt, false);
             if (userInfo == null)
             {
-                if (Logger.IsTraceEnabled) Logger.Trace("Token not found in DB");
-                return EmptyWithError("not-found");
+                if (Logger.IsTraceEnabled) Logger.Trace("User not found in DB");
+                return EmptyWithError("user-not-found");
```

Comment:
By disclosing this are we giving a malicious user more information than they need?

Reply:
Ok, I see

You wanted to hide the main reason of problem?

as I can see the secret which jwt token has been encrypted is unique for each user in portal, so at the end of day the attacker can find just its secret.

I think we can get more benefit for debug instead of users attacks.